tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/iterator/FederatedDescribeIteration.java

```diff
@@ -96,4 +96,13 @@ protected BindingSet convert(BindingSet sourceObject) throws QueryEvaluationExce
 			}
 		};
 	}
+
+	@Override
+	protected void handleClose() throws QueryEvaluationException {
+		try {
+			super.handleClose();
+		} finally {
+			queryInfo.close();
```

Comment:
Not sure if the queryInfo object should be closed here or not.

Reply:
I think here it is good to have. But even if not: I think there is a very much outer place for error handling of all query types which makes sure to abort in case of any errors